### ISSUE TITLE
/o/revoke_token/ ask for client_secret

### ISSUE THREAD

**Issue by rllola** [evonove/django-oauth-toolkit#244]
It seems that to revoke a token I need the client_secret and the client_id. When I try with only the client_id I have a 401 error.
I have been able to get an access token with only my client_id but I can't revoke it.
My app is declared as public and with resource owner password based.

```
POST /o/revoke_token/ HTTP/1.1
Host: 127.0.0.1:8000
Authorization: Bearer WQLwuZSHhsvOmHiAGPUqUVcDEaApl1
Cache-Control: no-cache
Postman-Token: 43bd0d57-9805-b3d1-93a0-6038d0ad12ce
Content-Type: application/x-www-form-urlencoded

token=3zT0A5y7LiKRamaeHNFubvJeC6h9w1&client_id=6iHKFZmNmG4GMOu25atyF8e5yAUZ9xdvbcODz7ld
```
The answer is { 'error': 'invalid_client' }

**Comment by flc** [comment]
Hey @rllola, I created an issue with the same problem a few days ago: https://github.com/evonove/django-oauth-toolkit/issues/242
You might want to take a look.
For now I create a client secret for the application and send it along with the client id.

**Comment by rllola** [comment]
Ok sorry I didn't see.
I have been talking about that with a coworker. He told me that the problem is actually in authlib which is use by DOT.
We are going to make a pull request for this issue.

**Comment by masci** [comment]
Added a test case to track this on #245

**Comment by masci** [comment]
The PR was merged in oauthlib.
Tests pass against the master, now waiting for a new release of oauthlib before closing this issue

**Comment by thenikso** [comment]
So this is fixed but not released? I also need to add the `client_secret` even if the app is marked as public. Any eta on the release? Thanks!

**Comment by masci** [comment]
Hi @thenikso this is a bug on the oauthlib side, it is fixed in the master but they haven't made a release yet and it seems to be no ETA, see:
https://github.com/idan/oauthlib/issues/346

**Comment by thenikso** [comment]
Thanks, I solved this on my side by making a custom "logout" endpoint with was in our specifications to begin with. I'll keep an eye on this anyway.

**Comment by mfc-julius** [comment]
+1

**Comment by mfc-julius** [comment]
Any progress with this?

**Comment by masci** [comment]
Should be fixed in 0.9.0

**Comment by raphael-riel** [comment]
This issue seems resolved and should be closed.
